site-scrapers/Walmart/index.js

```diff
@@ -0,0 +1,157 @@
+const { entityName, loginUrl } = require("./config");
+// const { parseAvailability, getStores } = require("./walmartBase");
+
+const moment = require("moment");
+
+module.exports = async function GetAvailableAppointments(
+    browser,
+    fetchService = liveFetchService()
+) {
+    console.log(`${entityName} starting.`);
+
+    const websiteData = await ScrapeWebsiteData(browser, fetchService);
+
+    const results = {
+        parentLocationName: `${entityName}`,
+        isChain: true,
+        timestamp: moment().format(),
+        individualLocationData: websiteData,
+    };
+
+    console.log(`${entityName} done.`);
+
+    return results;
+};
+
+/**
+ * Dependency injection: in live scraping, the fetchAvailability() in this module is used.
+ * In testing, mocks of these functions are injected.
+ */
+function liveFetchService() {
+    return {
+        getStores() {
+            return getStores();
+        },
+        async login(page) {
+            return await login(page);
+        },
+        async fetchStoreAvailability(page, storeNumber) {
+            return await fetchStoreAvailability(page, storeNumber);
+        },
+    };
+}
+
+async function ScrapeWebsiteData(browser, fetchService) {
+    const page = await browser.newPage();
+
+    const loginSucceeded = await fetchService.login(page);
+
+    const results = [];
+
+    const stores = fetchService.getStores();
+    if (loginSucceeded) {
+        Promise.all([
+            Object.keys(stores).forEach(async (storeNumber) => {
+                let response = await fetchService.fetchStoreAvailability(
+                    page,
+                    storeNumber
```

Comment:
Should this be `await Promise.all(...` ? (I'm not a Promise expert)

Reply:
My IDE flagged this during it's "code inspection"

Reply:
Not an expert either. Turns out that it's not needed, and cannot recall why I wrapped that code in the Promise.all. If I remove it, things work just fine.